unit/goto-programs/goto_program_declaration.cpp

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************\
+
+  Module: Unit tests for goto_program::validate
+
+  Author: Diffblue Ltd.
+
+ \*******************************************************************/
+
+#include <goto-programs/goto_function.h>
+#include <testing-utils/catch.hpp>
+#include <util/arith_tools.h>
+
+SCENARIO(
```

Comment:
We add a unit test for the decl checks here, but not for the dead checks which were also added in the same commit